### ISSUE TITLE
youtube-dl: switch to yt-dlp ? 

### ISSUE THREAD

**Issue by Vouivre** [kiss-community/community#781]
Since a few weeks now, the download speed for videos on youtube with `youtube-dl` is slower and it's limited at
about 50-60 kb/s. An issue has been raised recently, that a video was not possible to be downloaded. The last commit for `youtube-dl` was on 1.July. It should happen at some time, that it won't be possible to download
some videos. 

There are many issues about the download speed [like the following for example](https://github.com/ytdl-org/youtube-dl/issues/30097).

A fix is to use the fork [yt-dlp](https://github.com/yt-dlp/yt-dlp). I have tested and it works. 

I see two possibilities:

1. create a new package `yt-dlp`
2. We switch the source for `youtube-dl` to use `yt-dlp`. 

My favorite is the second choice. If `youtube-dl` becomes active again, we can switch back. If we have two packages, it doesn't solve every issues. For example `mpv` use `youtube-dl`. I expect it will be also slow. `ytfzf` use `youtube-dl`: `youtube-dl` is hardcoded in the code of `ytfzf`. So if we switch to `yt-dlp` and rename it to `youtube-dl`, it will fix also the problems for `mpv` and `ytfzf`.

@dilyn-corner  and other users: what do you think about it ?

**Comment by Riteo** [comment]
@Vouivre it still wouldn't be a drop-in replacement. In that case it might probably be better to make a shell script to run it with its compatibility options:

`youtube-dl`:
```
#!/bin/sh
yt-dlp --compat-options youtube-dl $@
```

But again, IMO it'd be simpler to just make a different package; they're different programs.

**Comment by Vouivre** [comment]
That's right. `yt-dlp` has other features, so it's not exactly the same. 

In the future, if the development of `youtube-dl` don't resume, other softwares depending on it will switch 
to another fork, probably this one. 

@dilyn-corner  do you agree with it ?

**Comment by Vouivre** [comment]
I think we can close this issue, probably @dilyn-corner will agree. 

Another pointer to create a separate package:  [a new version](https://github.com/mpv-player/mpv/releases/tag/v0.34.0)  of `mpv` has been released. By default, `mpv` now rely on `yt-dlp` instead of `youtube-dl`. 


From the release notes:

* ytdl_hook.lua: search for yt-dlp by default (preferred over youtube-dl)

This switch could also happen for other softwares.

**Comment by dilyn-corner** [comment]
Sad to hear about `youtube-dl`, glad to hear about `yt-dlp`! 

If `mpv` now relies on `yt-dlp`, I'm very tempted to say just drop `youtube-dl`... Can `tuitube` and `ytfzf` be used with `yt-dlp` (with `yt-dlp --compat-options youtube-dl` as @Riteo suggested above)?

If no real features are lost in using `yt-dlp` instead of `youtube-dl`, I'd probably just replace `youtube-dl` with this new project. 

@ioraff opened & closed #787 -- if you'd like keep `youtube-dl` around @Vouivre, @ioraff can reopen their PR. Unclear what to do with `tuitube` and `ytfzf` if `youtube-dl` or `yt-dlp` are build-time requirements, however; probably just test for either package before the build starts like how we build `cmake` (or used to build `libinput`).

**Comment by ioraff** [comment]
Reopened at #792, now with a `youtube-dl` compatibility script.

**Comment by Vouivre** [comment]
If there is a compatibility with `youtube-dl` I will probably drop `youtube-dl`. Otherwise I will keep it a few weeks to see if the developpment resume.

@ioraff: is it not possible to package `yt-dlp` like `youtube-dl` to have only one binary and not hundreds of python files ?

**Comment by ioraff** [comment]
@Vouivre: yes, it is. fixed.

**Comment by Vouivre** [comment]
@ioraff: thank you!

I installed `yt-dlp` but there was a checksum mismatch:

```
-> Building: explicit: yt-dlp 
-> Checking for pre-built dependencies 
-> yt-dlp Reading sources
found /home/macquat/.cache/kiss/sources/yt-dlp/2021.10.22.tar.gz
found /home/macquat/repos/perso/yt-dlp/files/youtube-dl
-> yt-dlp Verifying sources
- 63f267cb0f2e9d18afc8e51267be2a13ce60030eab7c04b25b94d6b90d2ba6d3
+ 63f267cb0f2e9d18afc8e51267be2a13ce60030eab7c04b25b94d6b90d2ba6d3
- 6fecba2ae65d9aad3c23a7c8d5326eb77934f6466db2c095456e4a2b524b145d
+ b1cd593518ca882681df9b546953c106aebbee27e78178f6c7f4ab6b2324f640
ERROR yt-dlp Checksum mismatch
```

Can you have a look at it ? Apparently it's for the file `youtube-dl`.

**Comment by ioraff** [comment]
@Vouivre fixed in #800. Thanks

**Comment by dilyn-corner** [comment]
Embarrassing for me :V :V :V
